src/main/java/org/testng/IConfigurationListener.java

```diff
@@ -38,4 +68,14 @@ default void onConfigurationSkip(ITestResult tr) {
   default void beforeConfiguration(ITestResult tr) {
     // not implemented
   }
+
+  /**
+   * Invoked before a configuration method is invoked.
+   *
+   * @param tr The test result
+   * @param tm The test method
+   */
+  default void beforeConfiguration(ITestResult tr, ITestNGMethod tm) {
+    // not implemented

```

Comment:
Should call `beforeConfiguration(ITestResult)` by default.

Reply:
@juherr - Then our builds will never pass and also the tests that I wrote will never be invoked. That is why I had to add that exception handling logic in the first place.

Reply:
Ok. The idea was to simplify the implementation by keeping only one method.

Reply:
@juherr  sure. got it. Let me work on them  and push the changes. Will ping you once I have done the changes.

Reply:
@juherr - I think the refactoring you suggest will not work for the following scenario :

```java
public class DummyClass implements IConfigurationListener {

  @Override
  public void beforeConfiguration(ITestResult tr) {
    System.err.println("First");
  }

  @Override
  public void beforeConfiguration(ITestResult tr, ITestNGMethod tm) {
    System.err.println("Second");
  }
}
```

Since we have overridden both the variants, our built in logic of having the two arg method call the one arg method as `default` implementation will not get called and `beforeConfiguration(tr)` will never get called. So I think the current changes are correct and dont need refactoring.

Reply:
Ok. Do you think we should deprecate the first variant?

Reply:
@juherr - I was also thinking about it.. but I don't know what impact it will have on build tools (Gradle already is an example and our own build breaks in some places due to it resorting to reflection to load our listeners)...